src/server/game/Entities/AreaTrigger/AreaTriggerTemplate.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AreaTriggerTemplate.h"
+
+AreaTriggerTypes AreaTriggerTemplate::GetType() const
+{
+    if (IsSphere())
+    {
+        return AREATRIGGER_TYPE_SPHERE;
+    }
+    else if (IsBox())
+    {
+        return AREATRIGGER_TYPE_BOX;
+    }
+    else if (IsPolygon())
+    {
+        return AREATRIGGER_TYPE_POLYGON;
+    }
+    else if (IsCylinder())
+    {
+        return AREATRIGGER_TYPE_CYLINDER;
+    }
+
+    return AREATRIGGER_TYPE_NONE;
+}
+
+// Init the MaxSearchRadius that will be used in TrinitySearcher, avoiding calculate it at each update
+void AreaTriggerTemplate::InitMaxSearchRadius()
+{
+    switch (GetType())
+    {
+        case AREATRIGGER_TYPE_SPHERE:
+        {
+            MaxSearchRadius = SphereDatas.Radius;
+            break;
+        }
+        case AREATRIGGER_TYPE_BOX:
+        {
+            MaxSearchRadius = std::max(std::max(BoxDatas.Extents[0], BoxDatas.Extents[1]), BoxDatas.Extents[2]);
```

Comment:
Don't take Extents[2] into account here, if the height is large you will make search visit unneccessarily large number of grids.
On top of that the equation should be `std::sqrt(BoxDatas.Extents[0] * BoxDatas.Extents[0] / 4 + BoxDatas.Extents[1] * BoxDatas.Extents[1] / 4)`